### PR TITLE
Solidity Optimizer options

### PR DESCRIPTION
**Pull request by moltam89**
## Description

Hey guys,

This PR is about the Solidity compiler's Optimizer.

It is enabled by default in the original scaffold-eth [repo](https://github.com/scaffold-eth/scaffold-eth/blob/fc2ae9213be121e9e813d0c6fcb02cbc21d3593f/packages/hardhat/hardhat.config.js#L283). 
I wasn't sure if I should just enable it for scaffold-eth-2 as well, or not. I decided to only add it as an option and update the readme, to let people know about this, before they deploy to a Live network.

Let me know what do you think.

Best,
Tamas

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

### CODE REVIEW

File: README.md
Comment (technophile-04) [inline README.md:1]: I think its fine we don't mention it in README, don't want people to do a side quest on `optimizer` while they are getting started 😅 so let's just remove this...We already provided useful links in `hardhat.config.ts` 🙌

**Review by technophile-04 — APPROVED** [review]
Tysm @moltam89 🙌,  cc @carletex , @sverps , @damianmarti for final thoughts before we merge this, here is the TLDR; -> https://github.com/scaffold-eth/scaffold-eth-2/pull/360#issuecomment-1581354916

**Comment by carletex** [comment]
I'm up for enabling it by default!

A nitpick:

```js
// By default, the Solidity Optimizer is enabled
// https://docs.soliditylang.org/en/latest/internals/optimizer.html#benefits-of-optimizing-solidity-code
// https://docs.soliditylang.org/en/latest/using-the-compiler.html#optimizer-options
// https://hardhat.org/hardhat-runner/docs/config
```

Isn't this too much? haha

`// By default, the Solidity Optimizer is enabled` This seems redundant (easily inferred from optimizer: { enabled: true} :D). And maybe we don't need 3 links? I feel that the important part is the "runs" parameter, so some helper text with a link could help there (move the https://docs.soliditylang.org/en/latest/using-the-compiler.html#optimizer-options link right above "runs"?)

Anyway, everything looks good, just suggesting some tiny tweaks.

Thanks!

**Comment by XOder64x** [comment]
After you fucked me, you want me to come back

lun., 12 iun. 2023, 19:11 Carlos Sánchez ***@***.***> a scris:

**Comment by technophile-04** [comment]
> // By default, the Solidity Optimizer is enabled This seems redundant (easily inferred from optimizer: { enabled: true} :D). And maybe we don't need 3 links? I feel that the important part is the "runs" parameter, so some helper text with a link could help there (move the https://docs.soliditylang.org/en/latest/using-the-compiler.html#optimizer-options link right above "runs"?)

Yes yes, I agree with this 

Tysm @moltam89 just pushed the changes Carlos suggested at https://github.com/scaffold-eth/scaffold-eth-2/pull/360/commits/aef0a990626f40c18088cc34776d7090158f348e 

Also tested that deployment and verification work correctly 🙌 

https://sepolia.etherscan.io/address/0x786b4Dff292876aD3aDE9CCD534373AEa1Ad8eD7#code

**Comment by moltam89** [comment]
Thank you guys!

**Pull request merged**